FirebaseStorage/Sources/StorageUploadTask.swift

```diff
@@ -87,6 +87,10 @@ import Foundation
       } else if let fileURL = self.fileURL {
         uploadFetcher.uploadFileURL = fileURL
         uploadFetcher.comment = "File UploadTask"
+
+        if Bundle.main.bundlePath.hasSuffix(".appex") {
+          uploadFetcher.useBackgroundSession = false
```

Comment:
The one issue with using `Bundle.main` is that it [will return the framework bundle](https://developer.apple.com/documentation/foundation/bundle/1410786-main) when Firebase Storage is being dynamically linked as opposed to statically linked.

Reply:
@mattcomi, are you dynamically or statically linking Firebase? My above comment may or may not be an issue here as I'm unfamiliar with app extensions.

Reply:
This check should be good. Could you please instead replace line 91 with a call to this API: https://github.com/google/GoogleUtilities/blob/26c898aed8bed13b8a63057ee26500abbbcb8d55/GoogleUtilities/Environment/Public/GoogleUtilities/GULAppEnvironmentUtil.h#L47?

When doing so, add 
```
s.dependency 'GoogleUtilities/Environment', '~> 7.12'
```

to the `FirebaseStorage.podspec` [here](https://github.com/firebase/firebase-ios-sdk/blob/9b661b01bff913eb937e55929c4cbd30229b6c2a/FirebaseStorage.podspec#L45url). 

And, the `Package.swift`'s target definition for Firebase Storage [here](https://github.com/firebase/firebase-ios-sdk/blob/9b661b01bff913eb937e55929c4cbd30229b6c2a/Package.swift#L1164).

Reply:
@ncooke3 Thanks so much for laying out those steps! Changes made 🙌